modules/PowermeterBSM/lib/transport.hpp

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2020 - 2021 Pionix GmbH and Contributors to EVerest
+
+#ifndef POWERMETER_TRANSPORT_HPP
+#define POWERMETER_TRANSPORT_HPP
+
+#include <modbus/modbus_client.hpp>
+
+#include "protocol_related_types.hpp"
+#include "known_model.hpp"
+
+#include <memory>
+
+/**
+ * Baseclass for transport classes.
+ *
+ * Transports are:
+ * - direct connection via modbus
+ * - connection via SerialComHub
+ */
+
+#include <generated/interfaces/serial_communication_hub/Interface.hpp>
+
+namespace transport {
+
+class AbstractDataTransport {
+
+public:
+    using Spt = std::shared_ptr<AbstractDataTransport>;
+
+    /**
+     * Starting at SunspecDataModelAddress fetch Types::SunspecRegisterCount register contents.
+     */
+    virtual transport::DataVector fetch(protocol_related_types::SunspecDataModelAddress,
+                                        protocol_related_types::SunspecRegisterCount) = 0;
+
+    /**
+     * fetch data from a KnownModel (example: Sunspec Common )
+     */
+    virtual transport::DataVector fetch(const known_model::AddressData& ad) = 0;
+
+    /**
+     * device specific: Trigger generation of a custom BSM signed snapshot.
+     */
+    virtual bool trigger_snapshot_generation_BSM() = 0;
+
+    /**
+     * device specific: Trigger generation of OCMF signed snapshot BSM power meter.
+     */
+    virtual bool trigger_snapshot_generation_BSM_OCMF() = 0;
+};
+
+/**
+ * data transport via modbus.
+ */
+class ModbusTransport : public AbstractDataTransport {
+
+protected:
+    everest::connection::SerialDeviceConfiguration m_cfg;
+    everest::connection::SerialDevice m_device;
+    // everest::connection::SerialDeviceLogToStream      m_device;
```

Comment:
Delete or keep?

Reply:
i would like to keep this as a reminder that the implementation with a logstream exists. This is usesful for debugging.